thejcpalma/palworld-dedicated-server-docker
# Update Variable Names and Enhance Error Handling

# Update Variable Names and Enhance Error Handling

## Summary

This pull request includes two commits:

1. Update `DAY_TIME_SPEEDRATE` to `DAY_TIME_SPEED_RATE` in Dockerfile and default.env. [`commit`](https://github.com/thejcpalma/palworld-dedicated-server-docker/commit/250b5143049b1ec7851680ad80c32d7bef19afe7)
2. Add an error log message when the RCON config file is not found. [`commit`](https://github.com/thejcpalma/palworld-dedicated-server-docker/commit/c4848872c0acf9e27dcf4f20bab4a329e90dd69d)
3. Update tag workflow

## Motivation and Context

The first commit ensures consistency by updating the variable name `DAY_TIME_SPEEDRATE` to `DAY_TIME_SPEED_RATE` in both Dockerfile and default.env files. This fixes daytime speed rate issues because the variable would always default to `1.000000` value when `SERVER_SETTINGS_MODE` was `auto`

The second commit addresses the need for better error handling by adding an error log message. When the RCON config file is not found, this log message enhances visibility and provides valuable information for debugging.

The third commit moves tags from the original commit to the one where the updated README.md file has the correct version so when consulting the tag the README.md shows the correct version instead of showing `dev`

## Description

### Update Variable Names
- Updated the variable name `DAY_TIME_SPEEDRATE` to `DAY_TIME_SPEED_RATE` in Dockerfile and default.env for consistency.

### Enhance Error Handling
- Added an error log message to be displayed when the RCON config file is not found. This improves the project's error handling and provides useful information for debugging.

### Update on Tag workflow
- Change logic to add the version tag to the commit where the README.md is changed and that way version shield in it stays concise with the tagged code.

## Testing Instructions

1. Build the Docker image.
2. Run the server with the new image.
4. Verify that the `DAY_TIME_SPEED_RATE` variable is correctly recognized.
5. Test the error scenario by intentionally removing the RCON config file.
6. Confirm that the error log message is displayed when the RCON config file is not found.

## Checklist
